design/open/ttv-wizard-mvp.md

```diff
@@ -0,0 +1,330 @@
+
+### Templating Service
+
+- **Endpoint**: `/api/v1/templates`  
+- **Request**:
+  - Method: `GET`
+  - Parameters:
+    - Template URL (`template_location`): `string` - retrieved from query string.
+    - Any other configurations required for the templates: `string` - retrieved from query string
+- **Response**:
+  - Return value: The expanded template
+  - Headers:
+    - `Content-Type` - The template's content type.
+- **Errors**:
+  1. *403- Forbidden*: The requesting user is forbidden from accessing the configurations or functionality (like generating credentials).
+  2. *400- Bad request*: The request is missing information necessary for the template’s expansion.
+  3. *500- Internal server error*: The lakeFS server cannot access some provided template locations.
+
+---
+
+## Possible Flows
+
+### Example template
+
+**Databricks Spark configurations**  
+Name: *databricksConfig.props.tt*
+```properties
+spark.hadoop.fs.s3a.impl=org.apache.hadoop.fs.s3a.S3AFileSystem
+{{with $creds := new_credentials}}
+spark.hadoop.fs.s3a.access_key={{$creds.Key}}
+spark.hadoop.fs.s3a.secret_key={{$creds.Secret}}
+{{end}}
+spark.hadoop.fs.s3a.endpoint={{ .querystring.lakefs_url }}
+spark.hadoop.fs.s3a.path.style.access=true
+```
+
+**Local Metastore configurations**
+Name: *localMetastoreConfig.xml.tt*
+```xml
+<configuration>
+    <property>
+        <name>fs.s3a.path.style.access</name>
+        <value>true</value>
+    </property>
+    <property>
+        <name>fs.s3.impl</name>
+        <value>org.apache.hadoop.fs.s3a.S3AFileSystem</value>
+    </property>
+    <property>
+        <name>fs.s3a.endpoint</name>
+        <value>{{ .querystring.lakefs_url }}</value>
+    </property>
+    {{with $creds := new_credentials}}
+    <property>
+        <name>fs.s3a.access.key</name>
+        <value>{{$creds.Key}}</value>
+    </property>
+    <property>
+        <name>fs.s3a.secret.key</name>
+        <value>{{$creds.Secret}}</value>
+    </property>
+    {{end}}
+</configuration>
+```
+
+### Happy flow - All steps
+
+1. The user clicks on ‘Create Repository’, then clicks on ‘Spark Quickstart’
+2. The wizard starts by showing an input textbox for the users to type the repo name. The user types ‘spark-repo’
+3. The wizard creates a repo named ‘spark-repo’ and sets a default ‘main’ branch.
+4. The wizard asks the users if they want to import existing data to lakeFS. The user specifies the location of the bucket (after they validated that the lakeFS role has the right permissions and that the bucket has the correct policy), and clicks ‘OK’.
+   * An object counter will show the progress of the import process and will signal once it’s over.
+5. The wizard asks the user for their lakeFS endpoint (and will show a default placeholder pointing to the current URL).
+6. The wizard will send a GET request to the templating service with a query string of the format:  
+    ```
+    ?lakefs_url=https://my-lakefs.io&template_location=databricksConfig.props.tt
+    ```
```

Comment:
Why can't the service infer the lakeFS url from the request itself? The UI knows the address of the server, right?
For trickier installation, where the address depends on the caller flow (i.e. with/without VPC), supplying the address gets harder for the user itself.

Reply:
The intention was that the URL will be set by default by the Wizard GUI (inferred from the URL it's working at), and we'll let the users change it if they want to.

> (and will show a default placeholder pointing to the current URL).